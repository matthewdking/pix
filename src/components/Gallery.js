import React, { Component } from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';

const GalleryContainer = styled.section`
  display: flex;
  align-items: center;
  justify-content: space-around;
  margin-top: 3rem;
`;

const GalleryWrapper = styled.div`
  text-align: center;
`;

const ImageWrapper = styled.div`
  margin: 1rem;
  display: inline-block;
  text-align: center;
`;

const ImageOne = styled.img`
  height: 25rem;
  width: 100%;
  margin: 0 auto;
`;

const Image = styled.img`
  height: 6rem;
`;

const ImageLink = styled.a``;

export class Gallery extends Component {
  showGallery = () => {
    if (this.props.pictures.fetched) {
      return (
        <GalleryWrapper>
          {this.props.pictures.pictures.map((picture, index) => {
            if (index === 0) {
              return (
                <ImageWrapper key={index}>
                  <ImageLink href={picture.pageURL}>
                    <ImageOne
                      src={picture.webformatURL}
                      alt={picture.pageURL}
                    />
                  </ImageLink>
                </ImageWrapper>
              );
            } else {
              return (
                <ImageWrapper key={index}>
                  <Image src={picture.previewURL} alt={picture.pageURL} />
                </ImageWrapper>
              );
            }
          })}
        </GalleryWrapper>
      );
    } else if (this.props.pictures.isFetching) {
      return <div>Loading...</div>;
    } else if (this.props.pictures.error) {
      return <div>Sorry the API appears to be down</div>;
    }
    return <div>Select a color to find some pictures</div>;
  };

  render() {
    return <GalleryContainer>{this.showGallery()}</GalleryContainer>;
  }
}

const mapStateToProps = state => ({
  pictures: state.pictures,
});

export default connect(mapStateToProps)(Gallery);
