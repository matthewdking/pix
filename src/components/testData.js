export const data = [
  {
    previewHeight: 99,
    likes: 11,
    favorites: 6,
    tags: 'cat, animal, mammal',
    webformatHeight: 426,
    views: 1014,
    webformatWidth: 640,
    previewWidth: 150,
    comments: 1,
    downloads: 669,
    pageURL: 'https://pixabay.com/en/cat-animal-mammal-portrait-pet-3134990/',
    previewURL:
      'https://cdn.pixabay.com/photo/2018/02/06/15/11/cat-3134990_150.jpg',
    webformatURL:
      'https://pixabay.com/get/ea34b20b21fd013ed95c4518b74b4f9fe471e6d404b014409df4c579a2e4b3_640.jpg',
    imageWidth: 6000,
    user_id: 124319,
    user: 'strecosa',
    type: 'photo',
    id: 3134990,
    userImageURL:
      'https://cdn.pixabay.com/user/2018/01/10/12-14-18-699_250x250.jpg',
    imageHeight: 4000,
  },
  {
    previewHeight: 99,
    likes: 6,
    favorites: 5,
    tags: 'cat, animal, mammal',
    webformatHeight: 426,
    views: 556,
    webformatWidth: 640,
    previewWidth: 150,
    comments: 1,
    downloads: 247,
    pageURL: 'https://pixabay.com/en/cat-animal-mammal-portrait-pet-3132568/',
    previewURL:
      'https://cdn.pixabay.com/photo/2018/02/05/16/19/cat-3132568_150.jpg',
    webformatURL:
      'https://pixabay.com/get/ea34b20d2df2093ed95c4518b74b4f9fe471e6d404b014409df4c579a2e4b3_640.jpg',
    imageWidth: 6000,
    user_id: 124319,
    user: 'strecosa',
    type: 'photo',
    id: 3132568,
    userImageURL:
      'https://cdn.pixabay.com/user/2018/01/10/12-14-18-699_250x250.jpg',
    imageHeight: 4000,
  },
  {
    previewHeight: 99,
    likes: 44,
    favorites: 47,
    tags: 'cat, tiger, tigerle',
    webformatHeight: 426,
    views: 5768,
    webformatWidth: 640,
    previewWidth: 150,
    comments: 3,
    downloads: 2811,
    pageURL:
      'https://pixabay.com/en/cat-tiger-tigerle-stripes-trouble-2356021/',
    previewURL:
      'https://cdn.pixabay.com/photo/2017/05/30/07/08/cat-2356021_150.jpg',
    webformatURL:
      'https://pixabay.com/get/eb36b40928f6003ed95c4518b74b4f9fe471e6d404b014409df4c579a2e4b3_640.jpg',
    imageWidth: 6000,
    user_id: 124319,
    user: 'strecosa',
    type: 'photo',
    id: 2356021,
    userImageURL:
      'https://cdn.pixabay.com/user/2018/01/10/12-14-18-699_250x250.jpg',
    imageHeight: 4000,
  },
  {
    previewHeight: 99,
    likes: 29,
    favorites: 31,
    tags: 'cat, tiger, tigerle',
    webformatHeight: 423,
    views: 2938,
    webformatWidth: 640,
    previewWidth: 150,
    comments: 1,
    downloads: 1408,
    pageURL:
      'https://pixabay.com/en/cat-tiger-tigerle-stripes-trouble-2356015/',
    previewURL:
      'https://cdn.pixabay.com/photo/2017/05/30/07/08/cat-2356015_150.jpg',
    webformatURL:
      'https://pixabay.com/get/eb36b40928f5043ed95c4518b74b4f9fe471e6d404b014409df4c579a2e4b3_640.jpg',
    imageWidth: 4889,
    user_id: 124319,
    user: 'strecosa',
    type: 'photo',
    id: 2356015,
    userImageURL:
      'https://cdn.pixabay.com/user/2018/01/10/12-14-18-699_250x250.jpg',
    imageHeight: 3235,
  },
  {
    previewHeight: 91,
    likes: 12,
    favorites: 21,
    tags: 'cat, mieze, animals',
    webformatHeight: 389,
    views: 1415,
    webformatWidth: 640,
    previewWidth: 150,
    comments: 0,
    downloads: 791,
    pageURL:
      'https://pixabay.com/en/cat-mieze-animals-cat-s-eyes-pride-2356013/',
    previewURL:
      'https://cdn.pixabay.com/photo/2017/05/30/07/08/cat-2356013_150.jpg',
    webformatURL:
      'https://pixabay.com/get/eb36b40928f5023ed95c4518b74b4f9fe471e6d404b014409df4c579a2e4b3_640.jpg',
    imageWidth: 5235,
    user_id: 124319,
    user: 'strecosa',
    type: 'photo',
    id: 2356013,
    userImageURL:
      'https://cdn.pixabay.com/user/2018/01/10/12-14-18-699_250x250.jpg',
    imageHeight: 3188,
  },
];
