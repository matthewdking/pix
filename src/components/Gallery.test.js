import React from 'react';
import renderer from 'react-test-renderer';
import { Gallery } from './Gallery';
import { data } from './testData';

const isFetchingProps = {
  pictures: [],
  fetched: false,
  isFetching: true,
  error: false,
};

const errorProps = {
  pictures: [],
  fetched: false,
  isFetching: false,
  error: true,
};

const fetchedProps = {
  pictures: data,
  fetched: false,
  isFetching: false,
  error: false,
};

const initialProps = {
  pictures: [],
  fetched: false,
  isFetching: false,
  error: false,
};

test('Gallery snapshot fetching', () => {
  const component = renderer.create(<Gallery pictures={isFetchingProps} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Gallery snapshot error', () => {
  const component = renderer.create(<Gallery pictures={errorProps} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Gallery snapshot fetched', () => {
  const component = renderer.create(<Gallery pictures={fetchedProps} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Gallery snapshot initial', () => {
  const component = renderer.create(<Gallery pictures={initialProps} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
