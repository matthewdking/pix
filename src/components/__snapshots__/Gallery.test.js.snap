// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery snapshot error 1`] = `
<section
  className="sc-bdVaJa iZoBPa"
>
  <div>
    Sorry the API appears to be down
  </div>
</section>
`;

exports[`Gallery snapshot fetched 1`] = `
<section
  className="sc-bdVaJa iZoBPa"
>
  <div>
    Select a color to find some pictures
  </div>
</section>
`;

exports[`Gallery snapshot fetching 1`] = `
<section
  className="sc-bdVaJa iZoBPa"
>
  <div>
    Loading...
  </div>
</section>
`;

exports[`Gallery snapshot initial 1`] = `
<section
  className="sc-bdVaJa iZoBPa"
>
  <div>
    Select a color to find some pictures
  </div>
</section>
`;
